controllers/pvc-mutator/storage-class-label-sync/sync.go

```diff
@@ -0,0 +1,55 @@
+package storageclasslabelsync
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/storageos/api-manager/internal/pkg/provisioner"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StorageClassParamsLabelSetter is responsible for setting labels
+// based on given StorageClass's params.
+type StorageClassParamsLabelSetter struct {
+	client.Client
+	log logr.Logger
+}
+
+// NewStorageClassParamsLabelSetter returns a new PVC label sync mutating admission
+// controller.
+func NewStorageClassParamsLabelSetter(k8s client.Client) *StorageClassParamsLabelSetter {
+	return &StorageClassParamsLabelSetter{
+		Client: k8s,
+		log:    ctrl.Log,
+	}
+}
+
+// MutatePVC mutates a given pvc with labels based on,
+// StoraeClass params.
+//
+// Errors returned here will block creation of the PVC.
```

Comment:
I think this would be better - I'm going to change this in the encryption PR too:
```
// Errors returned here may block creation of the PVC, depending on the
// FailurePolicy set in the webhook configuration.
```

Reply:
Does it true for `EncryptionKeySetter.MutatePVC()` too? At the moment i don't see the reason if not :)

Reply:
Yes, I've made the same wording change for encryption.